RecoLocalTracker/SiPhase2VectorHitBuilder/interface/VectorHitBuilderEDProducer.h

```diff
@@ -0,0 +1,42 @@
+//---------------------------------------------------------------------------
+// class VectorHitBuilderEDProducer
+// author: ebrondol,nathera
+// date: May, 2015
+//---------------------------------------------------------------------------
+
+#ifndef RecoLocalTracker_SiPhase2VectorHitBuilder_VectorHitBuilderEDProducer_h
+#define RecoLocalTracker_SiPhase2VectorHitBuilder_VectorHitBuilderEDProducer_h
+
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "RecoLocalTracker/SiPhase2VectorHitBuilder/interface/VectorHitBuilderAlgorithmBase.h"
+#include "DataFormats/Phase2TrackerCluster/interface/Phase2TrackerCluster1D.h"
+#include "DataFormats/TrackerRecHit2D/interface/VectorHit.h"
+
+#include <FWCore/ParameterSet/interface/ConfigurationDescriptions.h>
+#include <FWCore/ParameterSet/interface/ParameterSetDescription.h>
+
+class VectorHitBuilderEDProducer : public edm::stream::EDProducer<> {
```

Comment:
this should be moved to /plugins.
I see that there is also some curious choice of using this EDProducer in other contexts as an ESProduct, which should be verboten.